### PR TITLE
Improve flex ellipsis

### PR DESCRIPTION
**Pull request by wxiaoguang**
image

### CODE REVIEW

File: templates/devtest/label.tmpl [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	<div>
+		<h1>Label</h1>
+		<div>
+			<div class="flex-text-block tw-my-2">
+				<span class="ui label">simple label</span>
+				<span class="ui red label">red label</span>
+				<span class="ui green label">green label</span>
+			</div>
+			<div class="flex-text-block tw-my-2">
+				<span class="ui basic label">basic  label</span>
```
Comment (silverwind) [inline templates/devtest/label.tmpl]:
```suggestion
				<span class="ui basic label">basic label</span>
```

**Comment by silverwind** [comment]
I wonder if `tw-flex > * { min-width: 0}` would have any unexpected side-effects or if it would just make `gt-ellipsis` work in more such places.

**Comment by wxiaoguang** [comment]
> I wonder if `tw-flex > * { min-width: 0}` would have any unexpected side-effects or if it would just make `gt-ellipsis` work in more such places.

See the code: `min-width` should be set on the parent flex element, to limit its x-axis.

**Comment by wxiaoguang** [comment]
> or if it would just make `gt-ellipsis` work in more such places.

There won't be more places.

If we use `gt-ellipsis`, it should appear in a known element, for example, in a label, in a `flex-text-*`, etc. I won't expect to use `tw-flex` everywhere to contain `gt-ellipsis`, using `tw-flex` everywhere already makes a mess.

**Comment by silverwind** [comment]
> > I wonder if `tw-flex > * { min-width: 0}` would have any unexpected side-effects or if it would just make `gt-ellipsis` work in more such places.
> 
> See the code: `min-width` should be set on the parent flex element, to limit its x-axis.

No, it s[hould be on the flex children](https://css-tricks.com/flexbox-truncated-text/#aa-the-solution-is-min-width-0-on-the-flex-child), e.g.

```
tw-flex
  div <- min-width here
    div.gt-ellipsis

**Comment by wxiaoguang** [comment]
> No, it s[hould be on the flex children](https://css-tricks.com/flexbox-truncated-text/#aa-the-solution-is-min-width-0-on-the-flex-child), e.g.

Oh yes, my example is not accurate for this case. Will add more.

**Comment by wxiaoguang** [comment]
Let's see this example, the only `min-width` is on `.ui.label (flex)`:

image

**Comment by silverwind** [comment]
Maybe add some examples where you you three elements, and some with two. I would also name the page as "Ellipsis examples", not specific to labels.

**Comment by wxiaoguang** [comment]
> Maybe add some examples where you you three elements, and some with two. I would also name the page as "Ellipsis examples", not specific to labels.

Feel free to edit this PR.

**Review by delvh — APPROVED** [review]

**Review by silverwind — APPROVED** [review]
Guess it won't break anything at least.

**Comment by silverwind** [comment]
No backport, see https://github.com/go-gitea/gitea/pull/30344#discussion_r1564895591. Shouldn't have merged this. I really wonder why we didn't put the fixes into that PR instead.

**Comment by silverwind** [comment]
Partial revert: https://github.com/go-gitea/gitea/pull/30481

**Pull request merged**